Sources/Algorithms/Suffix.swift

```diff
@@ -0,0 +1,31 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Algorithms open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+//===----------------------------------------------------------------------===//
+// Suffix(while:)
+//===----------------------------------------------------------------------===//
+
+extension BidirectionalCollection {
+  @inlinable
+  public __consuming func Suffix(while predicate: (Element) throws -> Bool
+  ) rethrows -> [Element] {
+   var result = ContiguousArray<Element>()
+   self.reverse()
```

Comment:
I don’t believe `reverse()` can be used here (won’t compile), because it is a `mutating` function that reverses the elements of the collection in-place. This function, `suffix(while:)`, isn’t mutating (and should remain that way), so it can’t use `self.reverse()` in its implementation.

It seems that you might have meant to use the non-mutating version of that function, `reversed()`, which returns a reversed collection:

```swift
    for element in self.reversed() {
```

However, this comment likely won’t be applicable when changing the function to return `SubSequence` instead.